### PR TITLE
feat: simple Location model

### PR DESCRIPTION
**Pull request by raphodn**
### What

- new `Location` model
- it's goal is to store metadata fetched from OpenStreetMap, and link it to the Prices (next PR)

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -31,19 +31,44 @@ class User(Base):
     __tablename__ = "users"
 
 
+class Location(Base):
+    id = Column(Integer, primary_key=True, index=True)
+
+    osm_id = Column(BigInteger)
```
Comment (raphael0202) [inline app/models.py:37]: why don't we use osm_id as a primary key directly, as it's an INT?

File: app/models.py [hunk 2]
```diff
@@ -31,19 +31,44 @@ class User(Base):
...
+    osm_type = Column(ChoiceType(PriceLocationOSMType))
```
Comment (raphael0202) [inline app/models.py:38]: Not directly related to this PR, but we could rename PriceLocationOSMType into LocationOSMType (it doesn't add anything to add "Price")

File: app/schemas.py [hunk 3]
```diff
@@ -16,6 +16,26 @@ class UserBase(BaseModel):
...
+class LocationCreate(BaseModel):
+    model_config = ConfigDict(from_attributes=True, arbitrary_types_allowed=True)
+
+    osm_id: int = Field(gt=0)
+    osm_type: PriceLocationOSMType
+
+
+class LocationBase(LocationCreate):
+    id: int
+    osm_name: str | None
```
Comment (raphael0202) [inline app/schemas.py:28]: It's not mandatory, but it's not clear what each field is about. It's possible to describe the field using pydantic `description` parameter in `Field`.

**Review by raphael0202 — APPROVED** [review]

File: app/models.py (hunk 1 above)
Comment (raphodn) [reply app/models.py:37]: I'd rather not rely 100% on OSM. There's probably future locations that will not/never be in OSM. Like online stores, or temporary markets, or even closed locations 🤔...

File: app/schemas.py (hunk 3 above)
Comment (raphodn) [reply app/schemas.py:28]: i added this to #31

File: app/models.py (hunk 2 above)
Comment (raphodn) [reply app/models.py]: done here https://github.com/openfoodfacts/open-prices/pull/35/commits/2637cf778fed400dcf390ac4330b782deefdd7bb

**Pull request merged**